### ISSUE TITLE
Remove tests from cache

### ISSUE THREAD

**Issue by martinmladenov** [ciselab/TestGenie#134]
# Description
Depends on #101 

Tests that are no longer needed in the cache are removed from it.

# Definition of Done
* [x] Add logic for removing test cases to the data structure (Martin)
* [ ] When tests are applied, remove them from the cache
* [ ] Add a button next to each test case in the tool window to delete it from the cache
